insightsengineering/nesttemplate
# initialise the package

close https://github.com/insightsengineering/coredev-tasks/issues/61

Steps to test:

1. Install the newest pkgdown (in order to use bootstrap 5)
1. Install nesttemplate R package from that branch
1. Open any other NEST package
1. Modify its _pkgdown.yml file with the following:
  ```
  template:
    package: nesttemplate
  ```
1. Run pkgdown::build_site() for that package


To test:
- R package structure - DESCRIPTION, etc.
- README.md - whether it is clear enough
- template itself (located in inst/pkgdown/...) - esp. content and CSS styling (it has to be nice)


Implementation details:
- I created the [template](https://pkgdown.r-lib.org/articles/customise.html#template-packages) from scratch with the inspiration of https://github.com/ropensci-org/rotemplate and https://github.com/tidyverse/tidytemplate. I guess we can't just copy over one and change names.
- This is intended to be an easy-to-maintain, minimal template. We can enhance it and add some more sophisticated features in the future. Just wanted to init a repo and make use of it everywhere.
- I decided to change [bootswatch](https://bootswatch.com/) theme to the default one (in particular: don't use cerulean as the searchbar just looks super ugly).
- There is one TODO left regarding NEST homepage. I am working on that to have it clarified.
- I decided to put both NEST and pharmaverse as NEST belongs to the latter one. Happy to discuss that.
- If we finalise this PR then I will create a PR in all other repos as well as I will be asking https://github.com/orgs/insightsengineering/teams/idr to incorporate that in the workflow. That would also include upgrade pkgdown to v2.X.X (if not there yet). I don't think we need a staged deps functionalities for that repo. It should be treated as kind of external package. Happy to discuss it further. Some materials for thought: [pkgdown docs](https://pkgdown.r-lib.org/articles/customise.html#:~:text=If%20you%E2%80%99re%20building%20your%20site%20using%20GitHub%20actions), [dplyr entry in its DESCRIPTION file](https://github.com/tidyverse/dplyr/blob/main/DESCRIPTION#L77)


Results as of the moment of PR opening

- tern
before change:
![image](https://user-images.githubusercontent.com/12943682/154960089-ceb87e9a-a66d-4dd5-be0c-fc5b72b319b4.png)
after change:
![image](https://user-images.githubusercontent.com/12943682/154960233-fdff6b20-8321-4e8d-9b6a-d0f684e2e490.png)


- scda
before change:
![image](https://user-images.githubusercontent.com/12943682/154960143-53376215-28ba-45c1-a91d-08a7d991d1ae.png)
after change:
![image](https://user-images.githubusercontent.com/12943682/154960282-310cfe95-5365-460f-826d-2c876bfc6efb.png)
